provisioner/terraform/modules.go

```diff
@@ -0,0 +1,64 @@
+package terraform
```

Comment:
I'm curious about this.

We extract all other resources from the template by executing `terraform graph` (see `provisioner/terraform/executor.go` -> `graph()`) and then pass that to `provisioner/terraform/resources.go` -> `ConvertState()`. From there we retrieve all the template's resources, and they're then persisted.

Did this approach not work for modules?

Reply:
I couldn't find either module versions or their sources inside the state. This was the first approach I tried, and I switched to reading files only when I couldn't make it work.

Reply:
This is the output from a `terraform graph` run locally with a module specified:

```bash
$ terraform graph
digraph G {
  ...
  "module.jetbrains_gateway.coder_app.gateway" -> "coder_agent.main";
  "module.jetbrains_gateway.coder_app.gateway" -> "module.jetbrains_gateway.data.coder_parameter.jetbrains_ide";
  "module.jetbrains_gateway.coder_app.gateway" -> "module.jetbrains_gateway.data.coder_workspace.me";
  "module.jetbrains_gateway.coder_app.gateway" -> "module.jetbrains_gateway.data.coder_workspace_owner.me";
  "module.jetbrains_gateway.coder_app.gateway" -> "module.jetbrains_gateway.data.http.jetbrains_ide_versions";
}
```

Reply:
Yeah, the graph only displays the module names specified by the user in the template. It doesn’t show the source, like `registry.coder.com/modules/jetbrains-gateway/coder`, or the version, like `1.0.23`. That information is available in the files on disk.